### DISCUSSION TITLE
/api/v1/transaction-links api links incorrect transactions

### DISCUSSION THREAD

**Discussion by skynet2** [firefly-iii/firefly-iii#9058] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hi, 

I am using Firefly III api to automate some financial operations. 
When I use /api/v1/transaction-links it links incorrect transactions (looks like it uses previously created transaction instead of using id, which was specified)

For more details refer to Debug information.

TransactionID : 5154
![image](https://github.com/user-attachments/assets/b5633e8e-bff1-456c-ae17-ccb7a3966c1d)

TransactionID: 5153
![image](https://github.com/user-attachments/assets/d7c1a641-40a4-4870-94e2-a465cfae1255)

as we can see 5153 is linked to the transaction, but instead of 5154 its linked to 5152
![image](https://github.com/user-attachments/assets/ae23ee09-2fb9-48e4-90d2-8cffabf9fcf4)

Thanks,
Stas

### Debug information

/api/v1/transaction-links
Request:
```
{"link_type_id":"1","inward_id":"5154","outward_id":"5153","notes":"Commission"}
```
Response:
```
{"data":{"type":"transaction_links","id":"9","attributes":{"created_at":"2024-07-15T19:29:29+02:00","updated_at":"2024-07-15T19:29:29+02:00","inward_id":"5154","outward_id":"5153","link_type_id":"1","notes":"Commission"},"links":{"self":"https:\/\/ff.ggteam.xyz\/api\/v1\/transaction_links\/9","0":{"rel":"self","uri":"\/transaction_links\/9"}}}}
```

### Expected behaviour

Transactions with IDs 5154 and 5153 should be linked.

### Steps to reproduce

1. Use /api/v1/transactions to create two transactions. one Deposit & Withdrawal
2. Use /api/v1/transaction-links to link this transactions

**Comment by JC5** [comment, marked as answer]
It links transactions using the transaction journal ID, which is a different number. You can discover the transaction journal IDs using the API.

**Reply by skynet2** [reply]
Thanks @JC5 , it works.

I missed that note in the documentation. Thanks.
